src/syscall_sdl.c

```diff
@@ -47,38 +41,75 @@ typedef struct {
 } event_t;
 
 typedef struct {
-    event_t events[EVENT_QUEUE_CAPACITY];
-    size_t start, end;
-    bool full;
+    event_t *base;
+    size_t end;
 } event_queue_t;
 
+enum {
+    RELATIVE_MODE_SUBMISSION = 0,
+};
+
+typedef struct {
+    uint32_t type;
+    union {
+        union {
+            uint8_t enabled;
+        } mouse;
+    };
+} submission_t;
+
+typedef struct {
+    submission_t *base;
+    size_t start;
+} submission_queue_t;
+
 static SDL_Window *window = NULL;
 static SDL_Renderer *renderer;
 static SDL_Texture *texture;
+static uint32_t queues_capacity = 0;
```

Comment:
Add a blank line to separate SDL related variables and event queue specific ones. In addition, comments should appear as well.